nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/FetchAzureBlobStorage.java

```diff
@@ -49,10 +54,38 @@
 @SeeAlso({ ListAzureBlobStorage.class, PutAzureBlobStorage.class, DeleteAzureBlobStorage.class })
 @InputRequirement(Requirement.INPUT_REQUIRED)
 @WritesAttributes({
-    @WritesAttribute(attribute = "azure.length", description = "The length of the blob fetched")
+    @WritesAttribute(attribute = "azure.length", description = "The length of the blob fetched"),
+    @WritesAttribute(attribute = "azure.user.metadata.___", description = "If 'Write User Metadata' is set to 'True', the user defined metadata associated to the Blob object that is being listed " +
+        "will be written as part of the flowfile attributes")
 })
 public class FetchAzureBlobStorage extends AbstractAzureBlobProcessor {
 
+    protected static final PropertyDescriptor WRITE_USER_METADATA = new PropertyDescriptor.Builder()
```

Comment:
What if we move WRITE_USER_METADATA to AbstractAzureBlobProcesssor? Any reason to define in sub classes and repeat?

Reply:
No should be fine. Look into this on on Monday.
Btw Thanks for your review!

Reply:
Did move it to AzureStorageUtils. I think this Utils class is there because List and Fetch are not extended from the same base class. But maybe Blob should be moved too? Anyway not something for this Pull request. Is this fine?

Reply:
Looks good to me.